plugins/auth-backend/src/providers/github/provider.ts

```diff
@@ -78,12 +79,14 @@ export class GithubAuthProvider implements OAuthHandlers {
   private readonly tokenIssuer: TokenIssuer;
   private readonly catalogIdentityClient: CatalogIdentityClient;
   private readonly logger: Logger;
+  private readonly extraState: string | undefined;
 
   constructor(options: GithubAuthProviderOptions) {
     this.signInResolver = options.signInResolver;
     this.authHandler = options.authHandler;
     this.tokenIssuer = options.tokenIssuer;
     this.catalogIdentityClient = options.catalogIdentityClient;
+    this.extraState = options.extraState;
```

Comment:
If you default the value here, you wont have the complicated logic on line 113.

this.extraState = options.extraState | {}

Reply:
Good shout